salt/modules/vsphere.py

```diff
@@ -2039,17 +2040,19 @@ def list_datacenters(host, username, password, protocol=None, port=None):
         salt '*' vsphere.list_datacenters 1.2.3.4 root bad-password
 
     '''
-    service_instance = salt.utils.vmware.get_service_instance(host=host,
-                                                              username=username,
-                                                              password=password,
-                                                              protocol=protocol,
-                                                              port=port)
+    if any([host, username, password, protocol, port]):
```

Comment:
Should this be checking if the `service_instance is None and...`?

Either way here, even though the tests would be pretty simple, I'd feel better about having tests that specify which way this function should operate so it doesn't get accidentally changed in the future.